avm/src/daemon/daemon.rs

```diff
@@ -116,118 +128,157 @@ async fn post_v1_stats(
       .unwrap()
       .insert("clusterSecret".to_string(), json!(cluster_secret));
   } else {
-    error!(ErrorType::NoClusterSecret, "No cluster secret found.").await;
+    error!(NoClusterSecret, "No cluster secret found.").await;
   }
   Ok(post_v1("stats", stats_body).await)
 }
 
-async fn run_agz_b64(agz_b64: &str, priv_key_b64: &str, cert_b64: &str) -> DaemonResult<()> {
+async fn run_agz_b64(agz_b64: &str) -> DaemonResult<()> {
   let bytes = base64::decode(agz_b64);
   if let Ok(bytes) = bytes {
-    let agz = GzDecoder::new(bytes.as_slice());
-    let count = agz.bytes().count();
-    let mut bytecode = vec![0; count / 8];
-    let mut gz = GzDecoder::new(bytes.as_slice());
-    let gz_read_i64 = gz.read_i64_into::<LittleEndian>(&mut bytecode);
-    if gz_read_i64.is_ok() {
-      // TODO: return a Result in order to catch the error and log it
-      run(
-        bytecode,
-        HttpType::HTTPS(HttpsConfig {
-          port: 443,
-          priv_key_b64: priv_key_b64.to_string(),
-          cert_b64: cert_b64.to_string(),
-        }),
-      )
-      .await;
-    } else {
-      return Err("AGZ file appears to be corrupt.".into());
+    let pwd = env::current_dir();
+    match pwd {
+      Ok(pwd) => {
+        let priv_key = read(format!("{}/key.pem", pwd.display()));
+        let cert = read(format!("{}/certificate.pem", pwd.display()));
+        if let (Ok(priv_key), Ok(cert)) = (priv_key, cert) {
```

Comment:
Since the control port itself fails and exits the app, this is technically impossible, but supposing someone deleted the cert after spinning up the control port,  this `if let` could fail and the lack of an `else` on it means it would fall through to return an `Ok` while not actually spinning up the VM.

Reply:
Good catch, solved